script.js

```diff
@@ -1,4 +1,258 @@
+      mode = 'player 1 enter positions mode';
+      myOutputValue = 'Player 1,<br>Numbers : ' + player1DiceArray
+    + '<br> Positions : ' + dicePosition1Array + '<br>Please choose your positions';
+    } else {
+      myOutputValue = 'Please enter number of dice to use';
+    }
+    return myOutputValue;
+  }
+
+  // obtaining player 1's score
+  if (mode == 'player 1 enter positions mode') {
+    if (isNaN(Number(input)) == false && input != '' && input.length == player1DiceArray.length) {
+      generatePlayerScore(input);
+      player1Score = Number(playerScore);
+      player1TotalScore += player1Score;
+      myOutputValue = `Player 1, your score is ${player1Score}. It is now Player 2's turn. Player 2, 
+      please click submit`;
+      mode = 'player 2 variable dice mode';
+    } else {
+      myOutputValue = 'Please enter a valid choice';
+    }
+
+    // clear arrays for player 2's turn
+    dicePositionArray = [];
+    playerDiceArray = [];
+    return myOutputValue;
+  }
+
+  // obtaining player 2's dice array and positions array
+  if (mode == 'player 2 variable dice mode') {
+    rollNdice(player1DiceArray.length);
+    generatePositionArray(player1DiceArray.length);
+    console.log(generatePositionArray);
+    player2DiceArray = playerDiceArray;
+    dicePosition2Array = dicePositionArray;
+    mode = 'player 2 enter positions mode';
+    myOutputValue = 'Player 2,<br>Numbers : ' + player2DiceArray
+    + '<br> Positions : ' + dicePosition2Array + '<br>Please choose your positions';
+    return myOutputValue;
+  }
+
+  // obtaining player 2's score
+  if (mode == 'player 2 enter positions mode') {
+    if (isNaN(Number(input)) == false && input != '' && input.length == player2DiceArray.length) {
+      generatePlayerScore(input);
+      player2Score = Number(playerScore);
+      player2TotalScore += player2Score;
+      myOutputValue = `Player 2, your score is ${player2Score}<br>Player 1's score was ${player1Score}`;
+    } else {
+      myOutputValue = 'Please enter a valid choice';
+    }
+
+    // clear arrays for player 1's turn
+    dicePositionArray = [];
+    playerDiceArray = [];
+
+    // determining the winner
+    if (player1TotalScore > player2TotalScore) {
+      myOutputValue += `<br><br>Player 1's total score : ${player1TotalScore}
+      <br>Player 2's total score : ${player2TotalScore}<br><br> PLAYER 1 WINS !!<br><br>
+      Enter 'v' to continue, or 'n' to enter normal mode`;
+    } else if (player2TotalScore > player1TotalScore) {
+      myOutputValue += `<br><br>Player 1's total score : ${player1TotalScore}
+      <br>Player 2's total score : ${player2TotalScore}<br><br> PLAYER 2 WINS !!
+      <br><br>Enter 'v' to continue, or 'n' to enter normal mode`;
+    } else {
+      myOutputValue += `<br><br>Player 1's total score : ${player1TotalScore}
+      <br>Player 2's total score : ${player2TotalScore}<br><br> It's a DRAW !<br><br>
+      Enter 'v' to continue, or 'n' to enter normal mode`;
+    }
```

Comment:
Some elements of `main` can still be refactored into helper functions to reduce clutter and clarify logic in `main`